filter.go

```diff
@@ -0,0 +1,203 @@
+package stick
+
+import (
+	"math"
+	"strings"
+)
+
+func BuiltInFilters() map[string]Filter {
+	filters := make(map[string]Filter)
+	filters["abs"] = FilterAbs
+	filters["default"] = FilterDefault
+	filters["batch"] = FilterBatch
+	filters["capitalize"] = FilterCapitalize
+	filters["convert_encoding"] = FilterConvertEncoding
+	filters["date"] = FilterDate
+	filters["date_modify"] = FilterDateModify
+	filters["escape"] = FilterEscape
+	filters["first"] = FilterFirst
+	filters["format"] = FilterFormat
+	filters["join"] = FilterJoin
+	filters["json_encode"] = FilterJsonEncode
+	filters["keys"] = FilterKeys
+	filters["last"] = FilterLast
+	filters["length"] = FilterLength
+	filters["lower"] = FilterLower
+	filters["merge"] = FilterMerge
+	filters["nl2br"] = FilterNl2Br
+	filters["number_format"] = FilterNumberFormat
+	filters["raw"] = FilterRaw
+	filters["replace"] = FilterReplace
+	filters["reverse"] = FilterReverse
+	filters["round"] = FilterRound
+	filters["slice"] = FilterSlice
+	filters["sort"] = FilterSort
+	filters["split"] = FilterSplit
+	filters["striptags"] = FilterStripTags
+	filters["title"] = FilterTitle
+	filters["trim"] = FilterTrim
+	filters["upper"] = FilterUpper
+	filters["url_encode"] = FilterUrlEncode
+	return filters
+}
+
+func FilterAbs(ctx Context, val Value, args ...Value) Value {
+	n := CoerceNumber(val)
+	if 0 == n {
+		return val
+	}
+	return math.Abs(n)
+}
+
+func FilterBatch(ctx Context, val Value, args ...Value) Value {
+	// TODO: Implement Me
+	return val
+}
+
+func FilterCapitalize(ctx Context, val Value, args ...Value) Value {
+	s := strings.ToLower(val.(string))
```

Comment:
Anywhere we expect a string, we should call `CoerceString`, as this matches Twig and PHP's behavior of silently casting even with loss of data.